chia/wallet/wallet_state_manager.py

```diff
@@ -384,9 +384,10 @@ async def create_more_puzzle_hashes(
                 self.log.info(f"Done: {creating_msg} Time: {time.time() - start_t} seconds")
             await self.puzzle_store.add_derivation_paths(derivation_paths)
             if len(derivation_paths) > 0:
-                await self.wallet_node.new_peak_queue.subscribe_to_puzzle_hashes(
-                    [record.puzzle_hash for record in derivation_paths]
-                )
+                if wallet_id == self.main_wallet.id():
+                    await self.wallet_node.new_peak_queue.subscribe_to_puzzle_hashes(
+                        [record.puzzle_hash for record in derivation_paths]
+                    )
```

Comment:
This is the main change, only subscribing to newly derived puzzle hashes for wallet ID 1.  Everything else is fixing the wallet to be reliant on the correct way to pick up subscriptions of non XCH coins rather than relying on the secondary derived puzzle hash.

Reply:
given this is global, I'm surprised it's not affecting DID wallet, maybe it's not using derived puzzles anymore

Reply:
The DID uses derived puzzles for its own internal spend logic, but picks everything up via hint like it should. Most of the problems I fixed in this PR were related to change, and since singletons don't make change, they were unaffected.

Reply:
Does the DID wallet need to have correct state for its previously used derived puzzles in the wallet db to work correctly (after a fresh sync)?

Reply:
@aqk no, it should be able to resync itself, in the past versions we forgot to add a hint in a few cases and had some issues with it